Test container build is failing: error downloading openapi-generator-cli

**Describe the bug**
From today we started experiencing issue in the model registry container build.
The failure seems related to the download of the openapi-generator-cli tool during the Make `deps` target:
```
#20 [builder 14/14] RUN CGO_ENABLED=1 GOOS=linux GOARCH=amd64 make clean model-registry
#20 0.195 rm -Rf ./model-registry internal/ml_metadata/proto/*.go internal/converter/generated/*.go pkg/openapi
#20 0.202 protoc -I./api/grpc --go_out=./internal --go_opt=paths=source_relative \
#20 0.202 	--go-grpc_out=./internal --go-grpc_opt=paths=source_relative api/grpc/ml_metadata/proto/metadata_store.proto
#20 0.282 protoc -I./api/grpc --go_out=./internal --go_opt=paths=source_relative \
#20 0.282 	--go-grpc_out=./internal --go-grpc_opt=paths=source_relative api/grpc/ml_metadata/proto/metadata_store_service.proto
#20 0.427 openapi-generator-cli validate -i api/openapi/model-registry.yaml
#20 0.496 /workspace/bin/openapi-generator-installation/lib/node_modules/@openapitools/openapi-generator-cli/node_modules/@nestjs/common/file-stream/streamable-file.js:36
#20 0.496             this.options.length ??= bufferOrReadStream.length;
#20 0.496                                 ^^^
#20 0.496 
#20 0.496 SyntaxError: Unexpected token '??='
#20 0.496     at wrapSafe (internal/modules/cjs/loader.js:1029:16)
#20 0.496     at Module._compile (internal/modules/cjs/loader.js:1078:27)
#20 0.496     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
#20 0.496     at Module.load (internal/modules/cjs/loader.js:979:32)
#20 0.496     at Function.Module._load (internal/modules/cjs/loader.js:819:12)
#20 0.496     at Module.require (internal/modules/cjs/loader.js:1003:19)
#20 0.496     at require (internal/modules/cjs/helpers.js:107:18)
#20 0.496     at Object.<anonymous> (/workspace/bin/openapi-generator-installation/lib/node_modules/@openapitools/openapi-generator-cli/node_modules/@nestjs/common/file-stream/index.js:4:22)
#20 0.496     at Module._compile (internal/modules/cjs/loader.js:1114:14)
#20 0.496     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
#20 0.499 make: *** [Makefile:70: openapi/validate] Error 1
#20 ERROR: process "/bin/sh -c CGO_ENABLED=1 GOOS=linux GOARCH=amd64 make clean model-registry" did not complete successfully: exit code: 2
```

Issue reported here:
- https://github.com/opendatahub-io/model-registry/pull/267#pullrequestreview-1838413024
- https://github.com/opendatahub-io/model-registry/pull/270#pullrequestreview-1838383031

**To Reproduce**
Steps to reproduce the behavior:
1. Open a PR that triggers `Test container image build / build-image (pull_request)` GHA

**Expected behavior**
No failure

**Additional context**
We might have to refactor the way external tools are fetched/downloaded as we should also guarantee the container image is buildable in a disconnected environment

--- a/Dockerfile
+++ b/Dockerfile
@@ -9,6 +9,10 @@ COPY ["go.mod", "go.sum", "./"]
 RUN go mod download
 
 USER root
+# default NodeJS 14 is not enough for openapi-generator-cli, switch to Node JS currently supported
+RUN yum remove -y nodejs npm
+RUN yum module -y reset nodejs
+RUN yum module -y enable nodejs:18
 # install npm and java for openapi-generator-cli
 RUN yum install -y nodejs npm java-11
 

